### PR TITLE
Container contract updates

### PR DESCRIPTION
**Pull request by taeold**
Partial revert of https://github.com/firebase/firebase-tools/pull/4240 where we transformed EventFilters to list of arrays. We now prefer to keep the object structure.

To support path-pattern operator on event filters, we add new eventFilterPathPattern property. While we are at it, we also adds channel property to event trigger definition.

Note that most of the affected changes are in test files :/

Relevant SDK change: https://github.com/firebase/firebase-functions/pull/1070

### CODE REVIEW

File: src/deploy/functions/release/planner.ts [hunk 1]
```diff
@@ -196,10 +196,7 @@ export function changedV2PubSubTopic(want: backend.Endpoint, have: backend.Endpo
   if (have.eventTrigger.eventType !== v2events.PUBSUB_PUBLISH_EVENT) {
     return false;
   }
-
-  return (
-    backend.findEventFilter(have, "topic")?.value !== backend.findEventFilter(want, "topic")?.value
-  );
+  return have.eventTrigger.eventFilters.topic !== want.eventTrigger.eventFilters.topcic;
```
Comment (colerogers) [inline src/deploy/functions/release/planner.ts:199]: typo - `topcic` to `topic`

File: src/deploy/functions/backend.ts [hunk 2]
```diff
@@ -48,16 +48,7 @@ export interface CallableTriggered {
...
-
-// One or more event filters restrict the set of events delivered to an EventTrigger.
-interface EventFilter {
-  attribute: EventFilterAttribute;
-  value: string;
-
-  // if left unspecified, equality is used.
-  operator?: "match-path-pattern";
-}
+type EventFilterKey = "resource" | "topic" | "bucket" | "alerttype" | string;
```
Comment (colerogers) [inline src/deploy/functions/backend.ts:51]: while we're here, we can also add `appid` for alerts that use it

**Review by colerogers — CHANGES_REQUESTED** [review]
couple things, then we're g2g

**Review by colerogers — APPROVED** [review]

File: src/deploy/functions/backend.ts [hunk 3]
```diff
@@ -48,7 +48,7 @@ export interface CallableTriggered {
   callableTrigger: CallableTrigger;
 }
 
-type EventFilterKey = "resource" | "topic" | "bucket" | string;
+type EventFilterKey = "resource" | "topic" | "bucket" | "alerttype" | string;
```
Comment (inlined) [inline src/deploy/functions/backend.ts]: I wonder if it's silly to keep this type.

File: src/gcp/cloudfunctionsv2.ts [hunk 4]
```diff
@@ -485,6 +477,9 @@ export function functionFromEndpoint(endpoint: backend.Endpoint, source: Storage
   return gcfFunction;
 }
 
+/**
+ *
```
Comment (inlined) [inline src/gcp/cloudfunctionsv2.ts:481]: wat

**Review by inlined — COMMENTED** [review]
Woof. Lots of work all over again. TY for this.

**Pull request merged**